app/tools/android/test_data/extensions/contactextension/src/com/example/extension/ContactExtension.java

```diff
@@ -75,33 +75,29 @@ public void onMessage(int instanceID, String message){
 
     String readContact(String name) {
         Activity cActivity = mContext.getActivity();
-        // find the contacts by name
-        Cursor cursor = cActivity.getContentResolver().query(
-                ContactsContract.Contacts.CONTENT_URI, null,
-                ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'",
-                null, null);
-        if (cursor.moveToNext()) {
-            // get the first phone of the specified name
-            Cursor phones = cActivity.getContentResolver().query(
-                    ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
-                    null,
-                    ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'",
-                    null, null);
-
-            if (phones.moveToNext()) {
-                String phone = phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
-                phones.close();
-                cursor.close();
-                return phone;
-            } else {
-                Log.w(TAG, "no phone number for " + name);
-                cursor.close();
-                return "";
+        Cursor nameCursor = null;
+        Cursor phoneCursor = null;
+        String phoneNumber = "";
+        try {
+            // Find the contacts by name.
+            nameCursor = cActivity.getContentResolver().query(ContactsContract.Contacts.CONTENT_URI,
+                    null, ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'", null, null);
+            if (nameCursor != null && nameCursor.getCount() >= 1) {
+                // Get the first phone of the specified name.
+                phoneCursor = cActivity.getContentResolver().query(
+                        ContactsContract.CommonDataKinds.Phone.CONTENT_URI, null,
+                        ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'", null, null);
+                if (phoneCursor != null && phoneCursor.getCount() >= 1) {
```

Comment:
I think `if (phoneCursor != null && phoneCursor.moveToFirst())` is enough. `moveToFirst()` will return false if the cursor is empty